### PR TITLE
fix: retrieve package names output

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
fixes retrieves with package name's output to list the package name, path, and source retrieved into that dir
matches toolbelt with the package section and JSON
differs from toolbelt by including the `Retrieved Source` section

### What issues does this PR fix or reference?
@W-9584557@

### CODE REVIEW

File: src/commands/force/source/retrieve.ts [hunk 1]
```diff
@@ -105,12 +110,20 @@ export class Retrieve extends SourceCommand {
...
+    this.getFlag<string[]>('packagenames', []).forEach((name) => {
+      packages.push({ name, path: join(projectPath, name) });
+    });
+
     const formatterOptions = {
       waitTime: this.getFlag<Duration>('wait').quantity,
       verbose: this.getFlag<boolean>('verbose', false),
     };
     const formatter = new RetrieveResultFormatter(this.logger, this.ux, formatterOptions, this.retrieveResult);
+    formatter.packages = packages;
```
Comment (shetzel) [inline src/commands/force/source/retrieve.ts:126]: What do you think about changing the `formatterOptions` type to add an optional `packages` option instead?

**Review by shetzel — COMMENTED** [review]
there are a ton of changes in the yarn.lock file.  Maybe we can leave it alone for this PR and handle those updates when we do all the dependabot merges?

File: src/commands/force/source/retrieve.ts (hunk 1 above)
Comment (WillieRuemmele) [reply src/commands/force/source/retrieve.ts:126]: I was doing that at first, but that type is defined in the `ResultFormatter` class, which is a base class for `deploy` and `retrieve`, but `retrieve` only has packagenames. It felt awkward altering the base class for only one of the child classes... but I'm open to doing that way too

Comment (shetzel) [reply src/commands/force/source/retrieve.ts:126]: Why can't we do this:
```javascript
export interface RetrieveResultFormatterOptions extends ResultFormatterOptions {
  packages?: string;
}
```

and then:
`public constructor(logger: Logger, ux: UX, options: RetrieveResultFormatterOptions, result: RetrieveResult) {`

**Review by shetzel — APPROVED** [review]

**Pull request merged**